imports/plugins/core/payments/server/no-meteor/mutations/enablePaymentMethodForShop.js

```diff
@@ -0,0 +1,49 @@
+import ReactionError from "@reactioncommerce/reaction-error";
+import SimpleSchema from "simpl-schema";
+import { paymentMethods as allPaymentMethods } from "/imports/plugins/core/core/server/no-meteor/pluginRegistration";
+
+const paramsSchema = new SimpleSchema({
+  isEnabled: Boolean,
+  paymentMethodName: String,
+  shopId: String,
+  clientMutationId: { type: String, optional: true }
+});
```

Comment:
It's mixing concerns to have `clientMutationId` (which only GraphQL cares about) in here. You could either explicitly build the whole object argument without this property, or you can tell validation to ignore extra properties:

```js
paramsSchema.validate(input, { ignore: [SimpleSchema.ErrorTypes.KEY_NOT_IN_SCHEMA] });
```

Reply:
I went back and forth on this. The reason I landed where I did with having it passed is because `clientMutationId` is in the GQL schema itself. So I think your second suggestion is best: ignoring the field rather than defining it in the validation schema.